tests/test_applets/batchProcessing/test_batchProcessingGui.py

```diff
@@ -0,0 +1,224 @@
+        batch_processing_gui._data_role_widgets[role_names[0]].list_widget.addItems(filenames)
+
+        qtbot.mouseClick(batch_processing_gui.run_button, Qt.LeftButton, delay=1)
+        assert len(request_instances) == 1
+        assert_gui_in_running_state(batch_processing_gui)
+
+        # pretend we were successful:
+        req = request_instances.pop()
+        success_args, success_kwargs = req.notify_finished.call_args
+        assert len(success_args) == 1
+        success_args[0]()
+
+        assert_gui_in_init_state(batch_processing_gui)
+
+        qtbot.mouseClick(batch_processing_gui.run_button, Qt.LeftButton, delay=1)
+        assert len(request_instances) == 1
+        assert_gui_in_running_state(batch_processing_gui)
+
+        # cancel:
+        qtbot.mouseClick(batch_processing_gui.cancel_button, Qt.LeftButton, delay=1)
+        req = request_instances.pop()
+        req.cancel.assert_called_once()
+        cancel_args, cancel_kwargs = req.notify_cancelled.call_args
+        assert len(cancel_args) == 1
+
+        info_messagebox = mock.Mock()
+        with mock.patch(
+            "ilastik.applets.batchProcessing.batchProcessingGui.QMessageBox.information", new=info_messagebox
+        ):
+            cancel_args[0]()
+            info_messagebox.assert_called_once()
+
+        assert_gui_in_init_state(batch_processing_gui)
+
+        # failure:
+        qtbot.mouseClick(batch_processing_gui.run_button, Qt.LeftButton, delay=1)
+        assert len(request_instances) == 1
+        assert_gui_in_running_state(batch_processing_gui)
+
+        req = request_instances.pop()
+        failed_args, failed_kwargs = req.notify_failed.call_args
+        assert len(failed_args) == 1
+
+        critical_messagebox = mock.Mock()
+        with mock.patch(
+            "ilastik.applets.batchProcessing.batchProcessingGui.QMessageBox.critical", new=critical_messagebox
+        ):
+            failed_args[0](mock.MagicMock(Exception), None)
+            critical_messagebox.assert_called_once()
+
+        assert_gui_in_init_state(batch_processing_gui)
+
+
+def test_BatchProcessingGui_clear(qtbot, batch_processing_gui, role_names, filenames):
+    for role_name in role_names:
+        current_data_role_widget = batch_processing_gui._data_role_widgets[role_name]
+        current_data_role_widget.list_widget.addItems(filenames)
+        assert current_data_role_widget.list_widget.count() == len(filenames)
+
+    for role_name in role_names:
+        current_data_role_widget = batch_processing_gui._data_role_widgets[role_name]
+        qtbot.mouseClick(current_data_role_widget.clear_button, Qt.LeftButton, delay=1)
+        assert current_data_role_widget.list_widget.count() == 0
+
+
+does_not_raise = nullcontext()
+
+
+@contextmanager
+def assert_raises(exception_type):
+    with capture_exceptions() as exp:
+        yield
+        if not exp:
+            raise AssertionError("No exception raised")
+        elif exp[0][0] == exception_type:
```

Comment:
```suggestion
        elif issubclass(exp[0][0], exception_type):
```

Reply:
i had that first, but it actually returns the class, not the instance...

Reply:
Yes, since this is consistent with LSP and `pytest.raises`.

Reply:
allright ;)